cmds/bmp/update_state_command.go

```diff
@@ -45,10 +60,81 @@ func (cmd updateStateCommand) Options() cmds.Options {
 
 func (cmd updateStateCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if cmd.options.Server == "" {
+		return false, errors.New("cannot have empty server ID")
+	}
+
+	if cmd.options.State == "" {
+		return false, errors.New("cannot have empty state")
+	} else {
+		if !cmd.isValidState(cmd.options.State) {
+			return false, errors.New("the server state is incorrect!")
+		}
+	}
+
 	return true, nil
 }
 
 func (cmd updateStateCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s command: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
-	return 0, nil
+
+	cmd.ui.PrintlnInfo("WARNING: Be careful updating the state of a server, as it might break your deployment!")
+	if cmd.isConfirmed() {
+		updateStateResponse, err := cmd.bmpClient.UpdateState(cmd.options.Server, cmd.options.State)
+		if err != nil {
+			return 1, err
+		}
+
+		if updateStateResponse.Status != 200 {
+			return updateStateResponse.Status, nil
+		}
+
+		cmd.ui.PrintlnInfo("Update Successful!")
+
+		return 0, nil
+	} else {
+		return 1, errors.New("Update Cancelled!")
+	}
+}
+
+// Private Methods
+
+func (cmd updateStateCommand) isValidState(state string) bool {
+	validState := map[string]bool{
+		"bm.state.new":     true,
+		"bm.state.using":   true,
+		"bm.state.loading": true,
+		"bm.state.failed":  true,
+		"bm.state.deleted": true,
+	}
+
+	return validState[state]
+}
+
+func (cmd updateStateCommand) isConfirmed() bool {
+	var userInput string
+	cmd.ui.PrintfInfo("Continue to update? (type 'yes' to continue)")
+	_, err := cmd.ui.Scanln(&userInput)
+	if err != nil {
+		return false
+	}
+
+	if cmd.isYes(userInput) {
+		return true
+	}
+
+	return false
+}
+
+func (cmd updateStateCommand) isYes(userInput string) bool {
+	yes := map[string]bool{
```

Comment:
Can you move this and the other one as a constant or variable. Small optimization but seems would be something good to do and avoid creating and deleting these maps every time method call...